src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -407,7 +403,7 @@ private bool AcquireClientCredentials(ref byte[] thumbPrint)
                 }
                 else
                 {
-                    if (ClientCertificates.Count == 0)
+                    if (_clientCertificates == null || _clientCertificates.Count == 0)
```

Comment:
Ok, thanks. I wasn't planning to do any refactoring as part of this change. There's plenty of it that can and should be done later :)